central/sensor/service/connection/connection_test.go

```diff
@@ -63,6 +65,113 @@ func (s *testSuite) TestGetPolicySyncMsgFromPoliciesDoesntDowngradeBelowMinimumV
 	s.Equal(policyversion.CurrentVersion().String(), policySync.Policies[0].GetPolicyVersion())
 }
 
+func (s *testSuite) TestSendDeduperStateIfSensorReconciliation() {
+
+	cases := map[string]struct {
+		givenSensorCapabilities       []centralsensor.SensorCapability
+		givenSensorState              central.SensorHello_SensorState
+		assertReconciliationMapClosed bool
+		assertDeduperState            map[string]uint64
+		assertDeduperStateSent        bool
```

Comment:
I think, there is a mix of styles here:
- The `assertReconciliationMapClosed` is used as an expected value,
- The `assertDeduperState[Sent]` is used as a flag to enable or disable assertion and as an expected value.

I think that:
1. Naming the expectations as such - i.e.,`expectedReconciliationMapClosed` -  would make it clearer.
2. Naming `assertDeduperState` as `expectedDeduperState` to make it an expectation
3.  Remove `assertDeduperStateSent` in favour of `expectedDeduperState != nil`

It is not a blocker for merging, but a small cleanup could be beneficial here.

Reply:
I don't agree with (3) because it makes the test hide an important distinction: central could send empty deduper states. 

Renamed the assertion functions to have an `expect*` prefix